src/Features/Upscaling/FoveatedRender/Modes.cpp

```
@@ -23,7 +24,31 @@ namespace FoveatedRenderImpl
 
-	bool Core::ExecuteVRDlssCore(Streamline& streamline,
+	bool Core::DispatchUpscaleRegion(Streamline& streamline, uint32_t eyeIndex,
+		ID3D11Resource* colorIn, ID3D11Resource* colorOut, ID3D11Resource* depth, ID3D11Resource* mvec,
+		ID3D11Resource* reactiveMask, ID3D11Resource* transparencyMask,
+		uint32_t inW, uint32_t inH, uint32_t outW, uint32_t outH,
+		uint32_t fullEyeWidthIn, uint32_t fullEyeHeightIn)
+	{
+		auto& upscaling = globals::features::upscaling;
+		if (upscaling.GetUpscaleMethod() == Upscaling::UpscaleMethod::kFSR) {
+			// FSR3's motionVectorScale is a pixel extent (not DLSS's ratio-based
+			// correction from Bridge::ComputeMvecScale) that mvec values are multiplied
```

Cppcheck (2.21.0)

[error] Returning pointer to local variable 'fields' that will be invalid when returning.

(returnDanglingLifetime)